### PR TITLE
LOK-1787: stop minion if gateway return any certificate errors

### PR DESCRIPTION
**Pull request by freddy61025**
## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-1787

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: minion/minion-grpc/grpc-client/src/main/java/org/opennms/horizon/minion/grpc/Constant.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.horizon.minion.grpc;
+
+public class Constant {
```
Comment (cgorantla) [inline minion/minion-grpc/grpc-client/src/main/java/org/opennms/horizon/minion/grpc/Constant.java:31]: May be we could rename this to `GrpcErrorCodes`

File: minion/minion-grpc/grpc-client/src/main/java/org/opennms/horizon/minion/grpc/MinionGrpcClient.java [hunk 2]
```diff
@@ -240,6 +242,22 @@ public void onNext(RpcResponseProto value) {
...
+                    code = Constant.CA_PATH_ERROR;
+                } else if (rootCause instanceof CertificateExpiredException) {
+                    code = Constant.CERT_EXPIRED;
+                } else if (rootCause instanceof CertificateNotYetValidException) {
+                    code = Constant.CERT_NOTYET;
+                }
+                if (code != null) {
+                    LOG.error("{}. Going to shut down now.", rootCause.getMessage());
+                    handleDisconnect();
+                    System.exit(code);
```
Comment (cgorantla) [inline minion/minion-grpc/grpc-client/src/main/java/org/opennms/horizon/minion/grpc/MinionGrpcClient.java:258]: Did you consider using  [ systemService.halt](https://github.com/apache/karaf/blob/191abca836ae64f71ff11cb2a83d1d7e38a198a0/system/src/main/java/org/apache/karaf/system/commands/Shutdown.java#L69)

Also, it is better to have a wrapper Class to handle System related calls.

File: minion/minion-grpc/grpc-client/src/main/java/org/opennms/horizon/minion/grpc/Constant.java (hunk 1 above)
Comment (cgorantla) [reply minion/minion-grpc/grpc-client/src/main/java/org/opennms/horizon/minion/grpc/Constant.java:31]: or just `ErrorCodes`

File: minion/minion-grpc/grpc-features/src/main/feature/feature.xml [hunk 3]
```diff
@@ -49,6 +49,7 @@
         <feature>grpc-ratelimitter</feature>
         <feature>grpc-contract</feature>
         <feature>grpc-protobuf</feature>
+        <feature>system</feature>
```
Comment (cgorantla) [inline minion/minion-grpc/grpc-features/src/main/feature/feature.xml:52]: I think this should be available by default. Do we need to add this explicitly ?

**Review by cgorantla — APPROVED** [review]
LGTM!

File: minion/minion-grpc/grpc-features/src/main/feature/feature.xml (hunk 3 above)
Comment (freddy61025) [reply minion/minion-grpc/grpc-features/src/main/feature/feature.xml:52]: karaf-maven-plugin verify will fail if it is removed.

**Pull request merged**